### PR TITLE
[LW] Resolve TODOs

### PR DESCRIPTION
**Pull request by Jolyon-S**
**Goals (and why)**:
Now that the main piece of work is through, I wanted to resolve the TODOs sooner rather than later.

**Implementation Description (bullets)**:
Resolve TODOs. This is several small features, like changing things to be batched, adding necessary things for later, etc.

**Testing (What was existing testing like?  What have you done to improve it?)**:
Updated tests accordingly.

**Concerns (what feedback would you like?)**:
Not going to affect the wiring stuff.

**Where should we start reviewing?**:
`LockWatchValueCache`

**Priority (whenever / two weeks / yesterday)**:
This week.

### CODE REVIEW

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/ValueStoreImpl.java [hunk 1]
```diff
@@ -32,13 +32,13 @@
 import io.vavr.collection.HashSet;
 import java.util.stream.Stream;
 
-public final class ValueStoreImpl implements ValueStore {
+final class ValueStoreImpl implements ValueStore {
     // TODO(jshah): implement cache eviction based on cache size
```
Comment (Jolyon-S) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/ValueStoreImpl.java:36]: I think this should be a different PR.

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/LockWatchValueCacheImpl.java [hunk 2]
```diff
@@ -202,8 +210,17 @@ private void clearCache() {
     }
 
     private static Stream<CellReference> extractTableAndCell(LockDescriptor descriptor) {
-        // TODO(jshah): this has potentially large blow-up for cells with lots of zero bytes. We should probably make
-        //  users opt-in or warn when this is the case.
-        return AtlasLockDescriptorUtils.candidateCells(descriptor).stream();
+        List<CellReference> candidateCells = AtlasLockDescriptorUtils.candidateCells(descriptor);
```
Comment (gmaretic) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/LockWatchValueCacheImpl.java:213]: We can still pass in the actual schema and only try to match with the cached tables from the schema if this is something that we actually hit

**Review by gmaretic — APPROVED** [review]
LGTM

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/LockWatchValueCacheImpl.java (hunk 2 above)
Comment (Jolyon-S) [reply atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/LockWatchValueCacheImpl.java:213]: Let's leave that optimisation for now, but possibly something we can visit later.
